### ISSUE TITLE
t/jobspec/valid/basic.yaml is not valid jobspec

### ISSUE THREAD

**Issue by grondo** [flux-framework/flux-core#2993]
Stumbled across this recently.

`t/jobspec/valid/basic.yaml` contains:

```yaml
version: 1
resources:
  - type: slot
    count: 1
    label: foo
    with:
      - type: node
        count: 1
tasks:
  - command: [ "app" ]
    slot: foo
    count:
      per_slot: 1
attributes:
```

However, RFC14 at least requires that `attributes` be a object/dict, not unset or `null`.
This may cause a conforming jobspec parser to fail if it assumes `attributes` will be present, and will be an object.

However, I do notice that the jobspec schema explicitly allows `null`:
```
    "attributes": {
      "description": "system and user attributes",
      "type": ["object", "null"],
```

But [RFC 25](https://github.com/flux-framework/rfc/blob/master/spec_25.rst#attributes) says explicitly that
> The attributes key SHALL be a dictionary of dictionaries. The attributes dictionary MUST contain system key and MAY contain the user 

I'm not either of those MUST clauses are enforced at this time, by either the schema or JobspecV1 validator.

**Comment by grondo** [comment]
Also, it seems RFC 25 makes `attributes.system.duration` a required key, but this is not enforced by `validate_jobspec`.

**Comment by garlick** [comment]
Fixed by #2994?

**Comment by grondo** [comment]
Yes, sorry.
